telegram/ext/builders.py

```diff
@@ -0,0 +1,1338 @@
+
+        .. seealso:: `JobQueue <https://git.io/J0FCN>`_, `timerbot.py <https://git.io/J0FWf>`_
+
+        Note:
+            * :meth:`telegram.ext.JobQueue.set_dispatcher` will be called automatically by
+              :meth:`build`.
+            * The job queue will be automatically started and stopped by
+              :meth:`telegram.ext.Dispatcher.start` and :meth:`telegram.ext.Dispatcher.stop`,
+              respectively.
+            * When passing :obj:`None`,
+              :attr:`telegram.ext.ConversationHandler.conversation_timeout` can not be used, as
+              this uses :attr:`telegram.ext.Dispatcher.job_queue` internally.
+
+        Args:
+            job_queue (:class:`telegram.ext.JobQueue`, optional): The job queue. Pass :obj:`None`
+                if you don't want to use a job queue.
+
+        Returns:
+            :class:`DispatcherBuilder`: The same builder with the updated argument.
+        """
+        return self._set_job_queue(job_queue)  # type: ignore[return-value]
+
+    def persistence(
+        self: 'DispatcherBuilder[Dispatcher[BT, CCT, UD, CD, BD, JQ, PT], BT, CCT, UD, CD, BD, JQ, PT]',
+        persistence: InPT,
+    ) -> 'DispatcherBuilder[Dispatcher[BT, CCT, UD, CD, BD, JQ, InPT], BT, CCT, UD, CD, BD, JQ, InPT]':
+        """Sets a :class:`telegram.ext.BasePersistence` instance to be used for
+        :attr:`telegram.ext.Dispatcher.persistence`.
+
+        .. seealso:: `Making your bot persistent <https://git.io/J0FWM>`_,
+            `persistentconversationbot.py <https://git.io/J0FW7>`_
+
+        Warning:
+            If a :class:`telegram.ext.ContextTypes` instance is set via :meth:`context_types`,
+            the persistence instance must use the same types!
+
+        Args:
+            persistence (:class:`telegram.ext.BasePersistence`, optional): The persistence
+                instance.
+
+        Returns:
+            :class:`DispatcherBuilder`: The same builder with the updated argument.
+        """
+        return self._set_persistence(persistence)  # type: ignore[return-value]
+
+    def context_types(
+        self: 'DispatcherBuilder[Dispatcher[BT, CCT, UD, CD, BD, JQ, PT], BT, CCT, UD, CD, BD, JQ, PT]',
+        context_types: 'ContextTypes[InCCT, InUD, InCD, InBD]',
+    ) -> 'DispatcherBuilder[Dispatcher[BT, InCCT, InUD, InCD, InBD, JQ, PT], BT, InCCT, InUD, InCD, InBD, JQ, PT]':
+        """Sets a :class:`telegram.ext.ContextTypes` instance to be used for
+        :attr:`telegram.ext.Dispatcher.context_types`.
+
+        .. seealso:: `contexttypesbot.py <https://git.io/J0F8d>`_
+
+        Args:
+            context_types (:class:`telegram.ext.ContextTypes`, optional): The context types.
+
+        Returns:
+            :class:`DispatcherBuilder`: The same builder with the updated argument.
+        """
+        return self._set_context_types(context_types)  # type: ignore[return-value]
+
+
+class UpdaterBuilder(_BaseBuilder[ODT, BT, CCT, UD, CD, BD, JQ, PT]):
```

Comment:
Maybe inherit from `DispatcherBuilder`?

After all, all the setters of DispatcherBuilder are used by UpdaterBuilder

Reply:
We'd still have to change the docstrings & type hints and the latter would require a lot of `# type: ignores` …